nilearn/plotting/img_plotting.py

```diff
@@ -37,6 +39,43 @@
 # Core, usage-agnostic functions
 
 
+def check_threshold(thr, data):
+    """
+    Checks that the given threshold has an accepted string value
+
+    Parameters
+    ----------
+    thr: str or a list of strings or number or list of numbers
+        If threshold is a string itmush finish with a percent sign,
+        e.g. "99.7%", or if it a value it can be a real numbers.
+    data: a Numpy array of a map to calculate the
+        threshold using a string percentage value.
+
+    Returns
+    -------
+    returns threshold as an output also checks if the value entered
+    is valid
+
+    """
+    if isinstance(thr, _basestring):
+        message = ("If 'threshold' is given as string it "
+                   'should be a number followed by the percent'
+                   'sign, "e.g. 25.3%"')
+        if not thr.endswith('%'):
+            raise ValueError(message)
+        try:
+            percentile = float(thr[:-1])
+            thr = fast_abs_percentile(data, percentile) + 1e-6
+        except ValueError as exc:
+            exc.args += (message, )
+            raise
+        thr = thr
+    elif not isinstance(thr, numbers.Real):
+        raise TypeError('Threshold must be a real value but you '
```

Comment:
Why did you change the error message? I reckon the original one is more accurate:

```python
'edge_threshold should be either a number '
'or a string finishing with a percent sign'
```